src/tokens/hooks.ts

```diff
@@ -4,17 +4,34 @@ import useSelector from 'src/redux/useSelector'
 import {
   tokensByAddressSelector,
   tokensByCurrencySelector,
+  tokensByIdSelector,
   tokensListSelector,
+  tokensListWithAddressSelector,
 } from 'src/tokens/selectors'
 import { convertLocalToTokenAmount, convertTokenToLocalAmount } from 'src/tokens/utils'
 import { Currency } from 'src/utils/currencies'
 
-export function useTokenInfo(tokenAddress?: string | null) {
+/**
+ * @deprecated use useTokenInfo and select using tokenId
+ */
+export function useTokenInfoByAddress(tokenAddress?: string | null) {
   // TODO: Make this work with native tokens lacking a tokenAddress
   const tokens = useSelector(tokensByAddressSelector)
   return tokenAddress ? tokens[tokenAddress] : undefined
 }
 
+export function useTokenInfo(tokenId: string) {
+  return useSelector(tokensByIdSelector)[tokenId]
```

Comment:
We try to avoid chaining off selectors, but I can't remember exactly why, so this should be something like:

```TypeScript
export function useTokenInfo(tokenId: string) {
  const tokens = useSelector(tokensByIdSelector)
  return tokens[tokenId]
}

Reply:
I don't see anywhere else in the repo we chain off the useSelector hook. Maybe it's a style choice, but this isn't a pattern we've used in the repo previously.

Reply:
I'm not sure I understand the difference here besides style -- I've updated the usage to match your recommendation here regardless.